code/datums/wounds/_wounds.dm

```diff
@@ -266,23 +266,45 @@
 /// Setter for [victim]. Should completely transfer signals, attributes, etc. To the new victim - if there is any, as it can be null.
 /datum/wound/proc/set_victim(new_victim)
 	if(victim)
-		UnregisterSignal(victim, list(COMSIG_QDELETING, COMSIG_MOB_SWAP_HANDS, COMSIG_CARBON_POST_REMOVE_LIMB, COMSIG_CARBON_POST_ATTACH_LIMB))
-		UnregisterSignal(victim, COMSIG_QDELETING)
-		UnregisterSignal(victim, COMSIG_MOB_SWAP_HANDS)
-		UnregisterSignal(victim, COMSIG_CARBON_POST_REMOVE_LIMB)
+		UnregisterSignal(victim, list(
+			COMSIG_QDELETING,
+			COMSIG_MOB_SWAP_HANDS,
+			COMSIG_CARBON_POST_REMOVE_LIMB,
+			COMSIG_CARBON_POST_ATTACH_LIMB,
+			COMSIG_LIFE_WOUND_PROCESS,
+			COMSIG_LIFE_WOUND_STASIS_PROCESS,
+			SIGNAL_ADDTRAIT(TRAIT_STASIS),
+			SIGNAL_REMOVETRAIT(TRAIT_STASIS)))
 		if (actionspeed_mod)
 			victim.remove_actionspeed_modifier(actionspeed_mod) // no need to qdelete it, just remove it from our victim
+		remove_wound_from_victim()
 
-	remove_wound_from_victim()
 	victim = new_victim
 	if(victim)
 		RegisterSignal(victim, COMSIG_QDELETING, PROC_REF(null_victim))
-		RegisterSignals(victim, list(COMSIG_MOB_SWAP_HANDS, COMSIG_CARBON_POST_REMOVE_LIMB, COMSIG_CARBON_POST_ATTACH_LIMB), PROC_REF(add_or_remove_actionspeed_mod))
+		RegisterSignals(victim, list(
+			COMSIG_MOB_SWAP_HANDS,
+			COMSIG_CARBON_POST_REMOVE_LIMB,
+			COMSIG_CARBON_POST_ATTACH_LIMB), PROC_REF(add_or_remove_actionspeed_mod))
+		if(!HAS_TRAIT(victim, TRAIT_STASIS))
+			RegisterSignal(victim, COMSIG_LIFE_WOUND_PROCESS, PROC_REF(handle_process))
+		RegisterSignal(victim, SIGNAL_ADDTRAIT(TRAIT_STASIS), PROC_REF(wound_enter_stasis))
+		RegisterSignal(victim, SIGNAL_REMOVETRAIT(TRAIT_STASIS), PROC_REF(wound_exit_stasis))
 
 		if (limb)
 			start_limping_if_we_should() // the status effect already handles removing itself
 			add_or_remove_actionspeed_mod()
 
+/// If someone wants their wound to do something special in stasis, they need to manually
+/// implement a registration for processing while on stasis.
+/datum/wound/proc/wound_enter_stasis(datum/victim, trait)
+	SIGNAL_HANDLER
+	UnregisterSignal(src, COMSIG_LIFE_WOUND_PROCESS)
+
+/datum/wound/proc/wound_exit_stasis(datum/victim, trait)
+	SIGNAL_HANDLER
+	RegisterSignal(src, COMSIG_LIFE_WOUND_PROCESS, PROC_REF(handle_process))
+
```

Comment:
point of order, self registered signals are typically a bad idea because they can't be double registered. not fully relevant here just wanted to mention it